### PR TITLE
SQS build

### PR DESCRIPTION
**Pull request by kaydee0502**

### CODE REVIEW

File: devsnest/Gemfile [hunk 1]
```diff
@@ -67,3 +67,5 @@ gem 'newrelic_rpm', '~> 7.1'
 gem 'audited', '~> 4.9'
 
 gem 'leaderboard', '~> 3.12', '>= 3.12.1'
+
+gem 'aws-sdk-sqs', '~> 1.0.0.rc11'
```
Comment (vedansh) [inline devsnest/Gemfile:71]: new line

File: devsnest/app/controllers/api/v1/users_controller.rb [hunk 2]
```diff
@@ -103,6 +103,7 @@ def connect_discord
         return render_error({ message: message }) if temp_user.web_active?
 
         @current_user.merge_discord_user(temp_user.discord_id, temp_user)
+        @current_user.sqs_send_message()
```
Comment (vedansh) [inline devsnest/app/controllers/api/v1/users_controller.rb:106]: do it in a worker

Comment (vedansh) [reply devsnest/app/controllers/api/v1/users_controller.rb:106]: sqs_send_message will be a common function, pass parameters like message

File: devsnest/app/controllers/api/v1/users_controller.rb [hunk 3]
```diff
@@ -103,6 +103,7 @@ def connect_discord
         return render_error({ message: message }) if temp_user.web_active?
 
         @current_user.merge_discord_user(temp_user.discord_id, temp_user)
+        AwsSqsWorker.perform_async("verification", @current_user.id)
```
Comment (vedansh) [inline devsnest/app/controllers/api/v1/users_controller.rb:106]: are you sure we want to do this for all

**Review by LakshitXD — APPROVED** [review]

**Review by Yug-gurnani — APPROVED** [review]

File: devsnest/app/controllers/api/v1/users_controller.rb [hunk 4]
```diff
@@ -103,6 +103,8 @@ def connect_discord
         return render_error({ message: message }) if temp_user.web_active?
 
         @current_user.merge_discord_user(temp_user.discord_id, temp_user)
+        Event.generate(Event::VERIFIED, @current_user) if $sqs.present?
+        Event.generate(Event::WELCOME, @current_user) if $sqs.present?
```
Comment (vedansh) [inline devsnest/app/controllers/api/v1/users_controller.rb:107]: let's not send a message abhi

File: devsnest/config/routes.rb [hunk 5]
```diff
@@ -49,6 +49,9 @@
       jsonapi_resources :internal_feedback, only: %i[create update]
       jsonapi_resources :link, only: %i[show]
       jsonapi_resources :hackathon, only: %i[create index update show]
+      jsonapi_resources :notification_bot, only: %i[create index]
```
Comment (vedansh) [inline devsnest/config/routes.rb:52]: isme create kab call hoga?

**Comment by vedansh** [comment]
Fix codeclimate, add rspecs, let's push it

File: devsnest/config/routes.rb (hunk 5 above)
Comment (Adhikram) [reply devsnest/config/routes.rb]: Create nehi call hoga hum log abhi manually karenge .. galti seh gya h .. thik kar diya h

File: devsnest/app/controllers/api/v1/users_controller.rb (hunk 4 above)
Comment (Adhikram) [reply devsnest/app/controllers/api/v1/users_controller.rb]: Okey

**Review by Adhikram — APPROVED** [review]

**Pull request merged**